### ISSUE TITLE
menu only opens on release of $mod+c, not on press as expected

### ISSUE THREAD

**Issue by mielouk** [swaywm/sway#4625]
swaymsg -t get_version 
sway version 1.2 (Sep 26 2019, Arch Linux)

sway -d 2> ~/sway.log
https://8n1.org/16060/57bc on IRC #sway the user emersion said the keypresses are not visible in the log, which he finds weird. He also suggested to include  
libinput debug-events
https://8n1.org/16061/df19

* Configuration File:
https://8n1.org/16058/37be $mod is "Win/Super"

* Explanation

When calling the menu via $mod+c I've to be conscious about releasing $mod last, to let the menu appear. If I release 'c' last, I've to hit $mod again to let the menu appear. So it is each time a conscious effort to release $mod last to make it work the first time.

Is there a way to make it work on press rather than on release of $mod?

**Comment by dinnymate** [comment]
hmm. What menu is this? I'm using dmenu and it opens on press of c. It's the same with my terminal. When i press down the enter key it instantly opens.

**Comment by RedSoxFan** [comment]
Your debug log is using the wayland backend. What compositor are you running sway within? It looks like that is what's handling your binding.

**Comment by mielouk** [comment]
@dinnymate All other bindings work on first press as intended. This is bemenu, a menu for wayland, as dmenu is X only. [ArchWiki](https://wiki.archlinux.org/index.php/Sway#Application_launchers) [Reddit](https://www.reddit.com/r/swaywm/comments/b0svce/wayland_native_application_launcher/)

@RedSoxFan Not sure what you mean as 
> sway is an i3-compatible Wayland compositor
[SwayGitHub](https://github.com/swaywm/sway)

and I didn't change the default.

**Comment by Emantor** [comment]
@mielouk You are not starting sway from within a virtual console, your sway session seems to be started in a session which is already running a wayland compositor. This way the outer wayland compositor may eat the keypresses, since sway is handed the keypresses by the outer compositor. Also note that running sway from a login manager is [not supported](https://github.com/swaywm/sway/wiki#login-managers).

**Comment by mielouk** [comment]
No, I start sway after logging in on TTY1. In the past I tried it with SDDM, but that lead to problems, as you suggest. I've been logging in via TTY1 and then typing 'sway' for many weeks.
Could you point out what part of the log you are regarding to? I'm not aware of any extra steps.

**Comment by emersion** [comment]
```
2019-10-11 12:09:39 - [backend/wayland/backend.c:189] Creating wayland backend
```

**Comment by mielouk** [comment]
Isn't this setup by `BEMENU_BACKEND=wayland` which I've put in my .zshrc?

**Comment by emersion** [comment]
This line is printed by Sway. It means you're starting Sway from within Sway.

**Comment by mielouk** [comment]
How would I do that?
Here is my complete [config](https://8n1.org/16103/2a53) There is no extra call for sway, neither is there one in my .zshrc.

**Comment by mielouk** [comment]
What could be the reason for Sway being started from within Sway?

**Comment by mielouk** [comment]
Here is a [new log without sway being started from within sway](https://8n1.org/16104/1c71)
no `[backend/wayland/backend.c:189] Creating wayland backend` in there.
so the actions regarding the menu should be visible now. Sorry for my stupidity!
